datastream-utils/src/main/java/com/linkedin/datastream/common/LogUtils.java

```diff
@@ -92,4 +95,44 @@ public static String logSummarizedTopicPartitionsMapping(List<String> partitions
             .append(logNumberArrayInRange(topicPartitionsMap.get(topicName))))
         .collect(Collectors.joining(", "));
   }
+
+  /**
+   * prints one log line for each string smaller than size limit
+   * @param  message
+   */
+  public static void logStringsUnderSizeLimit(Logger log, String message, String contextPrefix, int part, double sizeLimit) {
+    int bufferAdjustedSizeLimit = (int) (sizeLimit - BUFFER_1KB);
```

Comment:
This calculation can be done only once when you'd have a user facing public method instead of having it in both the this location and the `isLessThanSizeLimit` method during each recursive call.